src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Markup/XamlReaderHelper.cs

```diff
@@ -456,6 +456,12 @@ internal bool IsXmlDataIsland()
         {
             return (_xmlDataIslandDepth != -1);
         }
+
+        internal bool CanInitializeCollectionFromString(Type type)
+        {
+            return IsACollection(type) && XamlTypeMapper.GetTypeConverterType(type) == null
+                && XamlTypeMapper.GetTypeConverterType(GetCollectionItemType(type)) != null;
+        }
```

Comment:
What you could test for though is whether the item type converter can convert from string

Reply:
But again you need to decide whether it is desirable to throw "no setter" error or converter error. Maybe the latter is better in this case too.